x-pack/plugins/endpoint/server/routes/resolver/queries/stats.ts

```diff
@@ -5,143 +5,143 @@
+        },
+        events: {
+          filter: { term: { 'event.kind': 'event' } },
+          aggs: {
+            ids: { terms: { field: 'endgame.unique_pid' } },
           },
         },
       },
-      index,
     };
   }
 
-  protected query(entityIDs: string[], index: string): JsonObject {
+  protected query(entityIDs: string[]): JsonObject {
     return {
-      body: {
-        size: 0,
-        query: {
-          bool: {
-            filter: [
-              { terms: { 'process.entity_id': entityIDs } },
-              {
-                bool: {
-                  should: [
-                    {
-                      bool: {
-                        filter: [
-                          { term: { 'event.kind': 'event' } },
-                          {
-                            bool: {
-                              must_not: {
-                                term: { 'event.category': 'process' },
-                              },
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            { terms: { 'process.entity_id': entityIDs } },
+            {
+              bool: {
+                should: [
+                  {
+                    bool: {
+                      filter: [
+                        { term: { 'event.kind': 'event' } },
+                        {
+                          bool: {
+                            must_not: {
+                              term: { 'event.category': 'process' },
                             },
                           },
-                        ],
-                      },
+                        },
+                      ],
                     },
-                    { term: { 'event.kind': 'alert' } },
-                  ],
-                },
+                  },
+                  { term: { 'event.kind': 'alert' } },
+                ],
               },
-            ],
-          },
-        },
-        aggs: {
-          alerts: {
-            filter: { term: { 'event.kind': 'alert' } },
-            aggs: {
-              ids: { terms: { field: 'process.entity_id' } },
             },
+          ],
+        },
+      },
+      aggs: {
+        alerts: {
+          filter: { term: { 'event.kind': 'alert' } },
+          aggs: {
+            ids: { terms: { field: 'process.entity_id' } },
           },
-          events: {
-            filter: { term: { 'event.kind': 'event' } },
-            aggs: {
-              ids: { terms: { field: 'process.entity_id' } },
-            },
+        },
+        events: {
+          filter: { term: { 'event.kind': 'event' } },
+          aggs: {
+            ids: { terms: { field: 'process.entity_id' } },
           },
         },
       },
-      index,
+    };
+  }
+
+  public formatResponse(response: SearchResponse<ResolverEvent>): StatsResult {
+    const alerts = response.aggregations.alerts.ids.buckets.reduce(
+      (cummulative: Record<string, number>, bucket: AggBucket) => ({
+        ...cummulative,
```

Comment:
:grey_question: Take this with a big grain of salt (b/c perf may not even matter that much here), but: https://jsperf.com/object-assign-vs-spreading

Reply:
Is object assign supposed to be faster? This is what's showing up for me:
![image](https://user-images.githubusercontent.com/56361221/82349918-b0fb7580-99c8-11ea-9474-7acd80025605.png)

Or maybe I'm misunderstanding.

Reply:
Why you should always take jsperf stuff with a grain of salt :rofl: But here's what mine shows:
on this one: https://jsperf.com/comparing-object-asign-spread/10
445K for .assign and 52K for spread

Reply:
Haha, gotcha.

Reply:
Speed reasons aside, there may be some (?) other benefit to having the iterator return the same object it receives (or maybe that's even worse, since you're mutating the object on each pass? Who knows... It's probably not worth testing, at any rate.)

Reply:
this is also the way the Kibana styleguide suggests to do it, so it's right all around